pkg/controller/replicator.go

```diff
@@ -76,6 +76,10 @@ func (r *replicator) RemoveBackend(address string) {
 
 	logrus.Infof("Removing backend: %s", address)
 
+	// Stop the monitoring goroutine in the Controller
+	monitorChan := backend.backend.GetMonitorChannel()
+	monitorChan <- nil
```

Comment:
Instead of directly writing against the monitoring channel we should instead consider calling.
backend.backend.StopMonitoring()

Since the above, will lead to a deadlock, for file based backends which return a nil channel.

Reply:
I changed it to StopMonitoring.